shell/browser/api/atom_api_web_contents.cc

```diff
@@ -1643,6 +1651,8 @@ void WebContents::Print(mate::Arguments* args) {
   options.Get("landscape", &landscape);
   settings.SetBoolean(printing::kSettingLandscape, landscape);
 
+  // We set the default to empty string here and only update
+  // if at the Chromium level if it's non-empty
```

Comment:
What happens when an empty device name is passed along ?

Reply:
https://github.com/electron/electron/pull/19598#discussion_r310895212 that's what is checked here; if no value is passed by the user it is set to the value initialized in `printing_context_->UseDefaultValues()` which is the system's default printer.

Reply:
Sorry for being unclear, I meant what happens when we pass along empty device string, does it still pick the default value or crash ? I am trying to understand if its necessary for an empty device name check.

Reply:
it errors if we don't do this, since it sets the printer to empty string which necessarily won't exist!

Reply:
Got it, thanks for clarifying.